sql/incoming-schema-changes.sql

```diff
@@ -127,3 +127,97 @@ values ('bu6mmul5vxci5vqc', 'kaw2cas7dyiq2tmg', 1);
 insert into playertimes (gameid, userid, time_in_minutes)
 values ('bu6mmul5vxci5vqc', 'pwamtkqtbeyc8eyn', 6);
 
+
+
+-- The roundMedianTime function takes the exact median time in minutes
+-- and rounds it. If the time is over an hour, round to the nearest 5 minutes.
+-- Otherwise, round to the nearest minute.
+
+DELIMITER $$
+
+CREATE FUNCTION roundMedianTime(
+    exact_median_in_minutes INT(5)
+)
+RETURNS INT(5)
+DETERMINISTIC
+BEGIN
+    DECLARE rounded_median_in_minutes INT(5);
+    IF exact_median_in_minutes > 60 THEN
+        SET rounded_median_in_minutes = (round(exact_median_in_minutes/5))*5;
+    ELSE
+        SET rounded_median_in_minutes = round(exact_median_in_minutes);
+    END IF;
+    RETURN (rounded_median_in_minutes);
+END $$
+
+DELIMITER ;
+
+
+-- View to calculate the estimated play time (the rounded median time) of each game
+CREATE VIEW `gametimes` AS 
+    SELECT 
+        DISTINCT gameid, 
+        roundMedianTime( median(time_in_minutes) OVER (PARTITION BY gameid) )  as `rounded_median_time_in_minutes`
+    FROM playertimes;
+
+
+-- Create a materialized view to store the data from the gametimes view
+CREATE TABLE gametimes_mv (
+  gameid VARCHAR(32) NOT NULL,
+  rounded_median_time_in_minutes INT(5) unsigned not null,
+  PRIMARY KEY (gameid)
```

Comment:
We should add a `KEY (rounded_median_time_in_minutes)` so we can search and sort by this.

Reply:
Added this.